compiler/rustc_const_eval/src/transform/promote_consts.rs

```diff
@@ -656,8 +655,7 @@ impl<'tcx> Validator<'_, 'tcx> {
 
         let is_const_fn = match *fn_ty.kind() {
             ty::FnDef(def_id, _) => {
-                is_const_fn(self.tcx, def_id)
-                    || is_unstable_const_fn(self.tcx, def_id).is_some()
+                self.tcx.is_const_fn_raw(def_id)
```

Comment:
why is this using raw? Won't that avoid feature gates?

Reply:
The original code is either stably-const or unstably-const, which can just be simplified.

Reply:
oh I see... this is a weird piece of code